File: python/02_selection_sort/selection_sort_2.py
class SelectionSort2:
    # This is the more traditional way of running selection sort. Use only one array and swap elements
    def selection_sort(self, arr):
        for i in range(len(arr)):
            smallest_index = i

            for j in range(i + 1, len(arr)):
                if arr[j] < arr[smallest_index]:
                    smallest_index = j

            arr[i], arr[smallest_index] = arr[smallest_index], arr[i]


if __name__ == "__main__":
    ss = SelectionSort2()
    my_list = [5, 3, 6, 2, 10, -5]

    # Don't have to return a list because the mutation happens on the original
    ss.selection_sort(my_list)
    print(my_list)

File: java/src/main/java/ch2_selection_sort/SelectionSort2.java
package ch2_selection_sort;

import java.util.Arrays;

public class SelectionSort2 {
    public static void main(String[] args) {
        int[] testArray = {5, 3, 6, 2, 10, -5};

        // Don't have to return a list because the mutation happens on the original
        selectionSort(testArray);
        System.out.println(Arrays.toString(testArray));
    }

    public static void selectionSort(int[] arr) {
        // More traditional way of selection sort - mutating the original array
        for (int i = 0; i < arr.length; i++) {
            int minIndex = i;

            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }

            if (minIndex != i) {
                int temp = arr[minIndex];
                arr[minIndex] = arr[i];
                arr[i] = temp;
            }
        }
    }
}
